api/src/main/java/org/pmiops/workbench/cdr/CdrVersionService.java

```diff
@@ -1,55 +1,50 @@
 package org.pmiops.workbench.cdr;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 import javax.inject.Provider;
-import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.AccessTierDao;
 import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
 import org.pmiops.workbench.db.model.DbCdrVersion;
-import org.pmiops.workbench.db.model.DbStorageEnums;
 import org.pmiops.workbench.db.model.DbUser;
 import org.pmiops.workbench.exceptions.ForbiddenException;
 import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.model.CdrVersionListResponse;
+import org.pmiops.workbench.model.CdrVersionMapResponse;
+import org.pmiops.workbench.model.CdrVersionMapResponseInner;
 import org.pmiops.workbench.model.DataAccessLevel;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CdrVersionService {
+  private final AccessTierDao accessTierDao;
+  private final AccessTierService accessTierService;
+  private final CdrVersionDao cdrVersionDao;
+  private final CdrVersionMapper cdrVersionMapper;
+  private final FireCloudService fireCloudService;
+  private final Provider<DbUser> userProvider;
 
-  private static final ImmutableSet<Short> REGISTERED_ONLY =
-      ImmutableSet.of(DbStorageEnums.dataAccessLevelToStorage(DataAccessLevel.REGISTERED));
-
-  private static final ImmutableSet<Short> REGISTERED_AND_PROTECTED =
-      ImmutableSet.of(
-          DbStorageEnums.dataAccessLevelToStorage(DataAccessLevel.REGISTERED),
-          DbStorageEnums.dataAccessLevelToStorage(DataAccessLevel.PROTECTED));
-
-  private static final ImmutableMap<DataAccessLevel, ImmutableSet<Short>>
-      DATA_ACCESS_LEVEL_TO_VISIBLE_VALUES =
-          ImmutableMap.<DataAccessLevel, ImmutableSet<Short>>builder()
-              .put(DataAccessLevel.REGISTERED, REGISTERED_ONLY)
-              .put(DataAccessLevel.PROTECTED, REGISTERED_AND_PROTECTED)
-              .build();
-
-  private Provider<DbUser> userProvider;
-  private Provider<WorkbenchConfig> configProvider;
-  private FireCloudService fireCloudService;
-  private CdrVersionDao cdrVersionDao;
+  private static final Logger log = Logger.getLogger(CdrVersionService.class.getName());
 
   @Autowired
   public CdrVersionService(
-      Provider<DbUser> userProvider,
-      Provider<WorkbenchConfig> configProvider,
+      AccessTierDao accessTierDao,
+      AccessTierService accessTierService,
```

Comment:
No need to inject `AccessTierDao` here and in the `AccessTierService`. I would remove accessTierDao and make a service method that returns `DbAccessTeir`

Reply:
good idea, thanks